app/frontend/courses/review/components/CoursesReview__Editor.res

```diff
@@ -1227,12 +1216,27 @@ let make = (
   let url = RescriptReactRouter.useUrl()
   let filter = Filter.makeFromQueryParams(url.search)
 
+  React.useEffect1(() => {
+    if state.nextSubmission == ReadyToLoad {
+      getNextSubmission(
+        send,
+        SubmissionDetails.courseId(submissionDetails),
+        filter,
+        nextSubmssionId => {
+          send(SetNextSubmissionId(nextSubmssionId))
+          send(SetNextSubmission(DataUnloaded))
+        },
+      )
+    }
+    None
+  }, [state.nextSubmission])
+
```

Comment:
I don't think you need the hook here; You have a controlled component which can directly call the API instead of setting the state here;  
```suggestion
```